fedbiomed/common/training_plans/_base_training_plan.py

```diff
@@ -568,25 +577,30 @@ def evaluate(data, target):
                 logger.critical(msg)
                 raise FedbiomedTrainingPlanError(msg) from exc
             # Log the computed value.
-            logger.debug(
-                f"Validation: Batch {idx}/{n_batches} "
-                f"| Samples {num_samples_observed_till_now}/{n_samples} "
-                f"| Metric[{metric_name}]: {m_value}"
-            )
-            # Further parse, and report it (provided a monitor is set).
-            if history_monitor is not None:
-                m_dict = self._create_metric_result_dict(m_value, metric_name)
-                history_monitor.add_scalar(
-                    metric=m_dict,
-                    iteration=idx,
-                    epoch=None,
-                    test=True,
-                    test_on_local_updates=(not before_train),
-                    test_on_global_updates=before_train,
-                    total_samples=n_samples,
-                    batch_samples=num_samples_observed_till_now,
-                    num_batches=n_batches
+            # Reporting
+
+            if idx % self.training_args()['log_interval'] == 0 or idx == 1 or idx == n_batches:
+                # FIXME: should we use MiniBatchTrainingIterationAccountant here ?
```

Comment:
Please create an issue as candidate to inspect this proposal in detail.

Reply:
Oh sorry, actually it was more for the review: what do you think about this condition statement? I slightly prefer what I did since it is "simple and stupid" (KISS), but one can use  `MiniBatchTrainingIterationAccountant` for a more advanced but complex solution

Reply:
I am not sure if it is compatible or not. It seems that `MiniBatchTrainingIterationAccountant` is designed for training and it accesses training plan attributes related to training such as `batch_size`, `num_updates` etc. It could be nice to make this class more generic in order to use it in testing or anywhere else but for now I think your implementation is fine. However, you can create an issue to propose to use `MiniBatchTrainingIterationAccountant`.

Reply:
If it is ok for you,  i will remove this comment. I don't think we will have the opportunity to improve these lines of code 
in the future

Reply:
Ok that's fine then.